src/mongo_types.ts

```diff
@@ -497,6 +502,16 @@ export type NestedPaths<Type> = Type extends
   : // eslint-disable-next-line @typescript-eslint/ban-types
   Type extends object
   ? {
-      [Key in Extract<keyof Type, string>]: [Key, ...NestedPaths<Type[Key]>];
+      [Key in Extract<keyof Type, string>]: Type[Key] extends Type
```

Comment:
@baileympearson any chance we could annotate these conditions with comments indicating the expected scenario to make it easier to follow?

Reply:
Done!